### ISSUE TITLE
Similarity Features are being shown in Metadata panel

### ISSUE THREAD

**Issue by tc-wleite** [sepinf-inc/IPED#1141]
@lfcnassif, in the master version, similarity features are being shown in the Metadata panel.
Is it intentional?
It uses a lot of space in the panel, and it is kind of an "internal" information (meaning not very useful to the user). 
I noticed that a while ago, but forgot to ask.

![image](https://user-images.githubusercontent.com/7217001/172545831-4a0d3587-f309-44ad-8de7-310828b2c31d.png)

**Comment by tc-wleite** [comment]
By the way, preparing for an idea for a new feature I have being thinking about in the last days, I may change a little bit the way the similarity features are stored.

**Comment by lfcnassif** [comment]
Hi @tc-wleite,

> Is it intentional?

It was. It is also being shown in table table, like the thumb byte[] array. We can hide that info from MetadataViewer if you think it is better.

**Comment by tc-wleite** [comment]
No, that is fine. I just wanted to check if it was intentional.
About the space consideration, there are other properties in some images that have even more data. I will try to see if there is a (not too complicated) way to make these huge property values "collapsible". I mean some general solution, whenever the content exceeds a certain number of characters, the first part would be shown by default, with a button to expand the area and show the full content.
The goal would be to avoid that these large values make harder to find other properties, while scrolling the metadata panel.
Of course that is a different issue...
I think we can close this one. Just let me know how the collapsible idea sounds to you.

**Comment by lfcnassif** [comment]
> Just let me know how the collapsible idea sounds to you.

I think it would be a very good enhancement!

**Comment by tc-wleite** [comment]
Closing as it is not a issue. May open another one later to deal with the collapsible idea.

**Comment by tc-wleite** [comment]
> > I may change a little bit the way the similarity features are stored
> 
> Would it be backwards compatible or should it be done before 4.0...?

Well, it could be compatible, but ideally it would break the compatibility.
It is not something critical, just a simple idea to use less space.
Do you think there is still time to include in 4.0?

**Comment by lfcnassif** [comment]
> Do you think there is still time to include in 4.0?

I think maybe a beta version could be out next week if we delay #180 to 4.0-final, but we could wait for and integrate the image similarity change if it is not that complex.

**Comment by tc-wleite** [comment]
It should be fast, but I still need to experiment a bit. I can definitely implement it (or come to the conclusion that it is not a good idea) in the beginning of the next week.

**Comment by lfcnassif** [comment]
Great, thank you @tc-wleite and take your time!

**Comment by lfcnassif** [comment]
Just a quick comment, looking closer at the screenshot, I thought it was being shown as base64 encoded byte[], but it isn't, we can  change the encoding to base64 to use less space.

**Comment by tc-wleite** [comment]
> Just a quick comment, I thought it was being shown as base64 encoded byte[], but looking closer at the screenshot, I see it isn't, we can change the encoding to base64 to use less space.

Looking at the code, byte arrays are being converted to hexadecimal representations, not base64.
https://github.com/sepinf-inc/IPED/blob/fbed03e0a5275dda0ef67bf908661f9268acfa9b/iped-viewers/iped-viewers-impl/src/main/java/dpf/sp/gpinf/indexer/ui/fileViewer/frames/MetadataViewer.java#L326-L328

**Comment by tc-wleite** [comment]
Changing to base64 makes sense, but still quite large.
I will include this minor change.

![image](https://user-images.githubusercontent.com/7217001/172834319-00629c3d-1681-40ff-aeb8-4cdd838943f2.png)
